pageserver/src/tenant/timeline.rs

```diff
@@ -132,6 +133,8 @@ use crate::virtual_file::{MaybeFatalIo, VirtualFile};
 use crate::walingest::WalLagCooldown;
 use crate::{ZERO_PAGE, task_mgr, walredo};
 
+const REL_SIZE_CACHE_CAPACITY: usize = 1024;
```

Comment:
I don't have good intuition on what the right size for this cache is. Therefore, I'd make this a tenant config since those are configurable at run-time. The data structures supports capacity resizing via `resize_capacity`.

This would give us a tool to address replicas with slow get pages due to rel size cache churn.